src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -2093,20 +2089,37 @@ internal override BaseClientSessionTransportManager CreateClientSessionTransport
                         StringUtil.Format(RemotingErrorIdStrings.KeyFileNotFound, this.KeyFilePath));
                 }
 
-                arguments = (string.IsNullOrEmpty(domainName)) ?
-                    string.Format(CultureInfo.InvariantCulture, @"-i ""{0}"" {1}@{2} -p {3} -s {4}", this.KeyFilePath, userName, hostname, this.Port, this.Subsystem) :
-                    string.Format(CultureInfo.InvariantCulture, @"-i ""{0}"" -l {1}@{2} {3} -p {4} -s {5}", this.KeyFilePath, userName, domainName, hostname, this.Port, this.Subsystem);
+                arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-i ""{0}""", this.KeyFilePath));
             }
-            else
+
+            // pass "-l login_name" commmand line argument to ssh if UserName is set
+            if (!string.IsNullOrEmpty(this.UserName))
             {
-                arguments = (string.IsNullOrEmpty(domainName)) ?
-                    string.Format(CultureInfo.InvariantCulture, @"{0}@{1} -p {2} -s {3}", userName, hostname, this.Port, this.Subsystem) :
-                    string.Format(CultureInfo.InvariantCulture, @"-l {0}@{1} {2} -p {3} -s {4}", userName, domainName, hostname, this.Port, this.Subsystem);
+                var parts = this.UserName.Split(Utils.Separators.Backslash);
+                if (parts.Length == 2)
+                {
+                    // ? pass DOMAIN\username as username@DOMAIN (still needed?)
+                    arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-l {0}@{1}", parts[1], parts[0]));
+                }
+                else
+                {
+                    arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-l {0}", this.UserName));
+                }
             }
 
+            // pass "-p port" command line argument to ssh if Port is set
+            if (this.Port != ReservedPort)
+            {
+                arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-p {0}", this.Port));
+            }
+
+            // pass "-s destination command" command line arguments to ssh where command is the subsystem to invoke on the destination
+            // ? note that ssh expects IPv6 addresses to not be enclosed in square brackets so trim them (still needed?)
+            arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-s {0} {1}", this.ComputerName.TrimStart('[').TrimEnd(']'), this.Subsystem));
+
             System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo(
                 filePath,
-                arguments);
+                string.Join(" ", arguments));
```

Comment:
Please use new `ProcessStartInfo` ArgumentList property to pass the argument list.

Reply:
I've used ArgumentList.

Reply:
Hrm. The Unix-specific StartSSHProcess parses startInfo.Arguments into an argv.

https://github.com/PowerShell/PowerShell/blob/f1b96eb74f957a9fea66478d03da45dd9d849e65/src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs#L2206

To use ArgumentList rather than Arguments, I need to modify the Unix-specific StartSSHProcess, which starts to expand the impact of this pull request.

Please advise.

Reply:
The smallest change could be in ParseArgs:

from

```C#
var argsToParse = psi.Arguments.Trim();
```

to

```C#
var argsToParse = String.Join(" ", psi.ArgumentList).Trim();
```

If acceptable, I'll commit and push.